### ISSUE TITLE
New rule: identify missing PropType definition

### ISSUE THREAD

**Issue by joetidee** [yannickcr/eslint-plugin-react#1887]
I accidentally forgot to add the `PropTypes` assignment to a prop and this was not picked up by eslint:

```
const propTypes = {
    actions: {
        something: PropTypes.func.isRequired,
    }.isRequired,
};
```

should have been

```
const propTypes = {
    actions: PropTypes.shape({
        something: PropTypes.func.isRequired,
    }).isRequired,
};
```

**Comment by ljharb** [comment]
I'd expect the first example to give you a missing propType warning for "actions"

**Comment by joetidee** [comment]
it didn't.

**Comment by joetidee** [comment]
It's an interesting question because the error does throw up in the browser console so React is recognising that the property has not been correctly assigned a `PropTypes.*`. So, should eslint allow us developers to catch this in the build pipelines before it gets to production - i think yes. Currently, there isn't a rule that could cover this, so perhaps we need a rule called `no-proptype-assigment` ?

**Comment by joetidee** [comment]
Ah I see. In which case if custom PropTypes have to take the form of a function, then I guess, yes, it is detecting that it is an object literal and throwing an error of incorrect-proptype
